x-pack/plugins/fleet/server/services/agents/upgrade.ts

```diff
@@ -89,19 +105,40 @@ export async function sendUpgradeAgentsActions(
             showInactive: false,
           })
         ).agents;
-  const agentsToUpdate = options.force
+
+  // upgradeable if they pass the version check
+  const upgradeableAgents = options.force
     ? agents
     : agents.filter((agent) => isAgentUpgradeable(agent, kibanaVersion));
+
+  // get any policy ids from upgradable agents
+  const policyIdsToGet: string[] = upgradeableAgents
```

Comment:
maybe try to have unique policy id here? otherwise I think it's going to fetch multiple time the same policy

Reply:
@nchaulet I added this in https://github.com/elastic/kibana/pull/91303/commits/0f967d7bedec8f9b0153b4135a08b556539f7af2